### ISSUE TITLE
Polygon.contains with Segment returns wrong result

### ISSUE THREAD

**Issue by gigerdo** [alexbol99/flatten-js#65]
I have the following geometry:
* A polygon with a hole
* A segment crossing that hole
![image](https://user-images.githubusercontent.com/204755/98397224-4f56c280-205f-11eb-913a-d6fe430963b6.png)

Expected result: 
polygon.contains(segment) == false

Actual result:
For this situation, Polygon.contains returns true, even though the segment is not completely contained in the polygon (only the end points). 

Here is an observable showing the situation:
https://observablehq.com/@giger/flattenjs-problem-with-covered

**Comment by gigerdo** [comment]
As a sidenote, I noticed that Flatten.Relations.relate is not defined for the parameter combination
(Polygon, Segment), but is defined for (Segment, Polygon). Is the order of parameters relevant for this function?

**Comment by gigerdo** [comment]
Another case I found for `Flatten.Relations.touch`. With this specific polygon the method returns `true` even tough the segment goes straight through the polygon. `Relations.intersect` works fine in this case.

```
const poly = new Flatten.Polygon([
    Flatten.point(710,750),
    Flatten.point(688.9918693812442,814.656377752172),
    Flatten.point(633.9918693812442,854.6162167924668),
    Flatten.point(566.0081306187558,854.616216792467),
    Flatten.point(511.0081306187558,814.656377752172),
    Flatten.point(490,750),
    Flatten.point(511.0081306187558,685.343622247828),
    Flatten.point(566.0081306187558,645.3837832075332),
    Flatten.point(633.9918693812442,645.383783207533),
    Flatten.point(688.9918693812442,685.343622247828),
]);
const seg = segment(point(600, 900), point(620, 570));

Flatten.Relations.touch(seg, poly); // === true, but should be false
Flatten.Relations.intersect(seg, poly); // === true, which is correct
```

**Comment by alexbol99** [comment]
Hi @gigerdo ,

Sorry for big delay with this issue.
Bug fixed in v1.2.22, see https://next.observablehq.com/d/ea7cc749d4dd0158

Thanks,
Alex Bol

**Comment by gigerdo** [comment]
No worries, thanks a lot for fixing it! It is a great library 👍
